rdmo/projects/assets/js/api/ProjectsApi.js

```diff
@@ -10,7 +10,12 @@ class ProjectsApi extends BaseApi {
       if (response.ok) {
         return response.json()
       } else {
-        throw new Error(response.statusText)
+        return response.json().then(errorBody => {
```

Comment:
You could throw a custome error class here, like `CatalogNotFoundError` and check for this in the action. A bit like its done in https://github.com/rdmorganiser/rdmo/blob/dev-2.2.0-projects-pagination/rdmo/core/assets/js/api/BaseApi.js#L42

Reply:
I don't understand what you mean.
To throw a CatalogNotFoundError I would have to implement logic again in the API call (to find out if it's such an error) 
I thought we agreed not to do so.
And would you expect the new class to reside in the ProjectsApi or the BaseApi?

Reply:
In ProjectsAPI like here https://github.com/rdmorganiser/rdmo/blob/dev-2.2.0-projects-pagination/rdmo/core/assets/js/api/BaseApi.js#L14 then you can just check `if (error.constructor.name === 'CatalogNotFoundError')`.

Reply:
Ah forget it, you need to check for the catalog anyway.

Reply:
You could just use the `ValidationError` from BaseAPI if you want.

Reply:
I think this only makes sense if I check in the API call if it's a catalog error.... 
And then in the actions file I would only check if it is such an error. Otherwise that makes no sense to me.

Reply:
Ok, now that I used a defined error class I understand better. I'll try that

Reply:
I call it BadRequestError